File: app/dbmodels/SoftwareVersion.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'SoftwareVersion',
    'menu'          => ''
);
$table['fields'] = array(
    'id'                        => DBModels::type('increments',
                                                  array('visible' => false)),
    'name'                      => DBModels::type('string'),
    'comment'                   => DBModels::type('text'),
    'software_id'               => DBModels::type('integer'),
    'states_id'  => DBModels::type('integer'),
    'operatingsystems_id'   => DBModels::type('integer'),
);

$table['relationships'] = array(
    'assets'     => array(
        'type'  => 'belongsToMany',
        'item'  => 'Asset',
        'linktable' => 'installed_software_versions',
        'linkfields' => array('is_deleted', 'is_dynamic', 'is_deleted_asset', 'is_template_asset')),
    'software' => array(
        'type' => 'belongsTo',
        'item' => 'Software'),
    'state' => array(
        'type' => 'belongsTo',
        'item' => 'State'),
    'operatingsystem' => array(
        'type' => 'belongsTo',
        'item' => 'OperatingSystem'),
);

include "_commonentitylink.php";
include "_commonrecursiveentitylink.php";

File: fusionglpi/database/software_version.py
from fusionglpi import db

class SoftwareVersion(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    name = db.Column(db.String(255, None, True))
    comment = db.Column(db.Text)
    software_id = db.Column(db.Integer, db.ForeignKey('software.id'))
    state_id = db.Column(db.Integer, db.ForeignKey('state.id'))
    operatingsystem_id = db.Column(db.Integer, db.ForeignKey('operatingsystem.id'))

    installed_software_versions = db.relationship('InstalledSoftwareVersion', backref='software_version')
